nixos/modules/services/databases/cassandra.nix

```diff
@@ -38,13 +38,18 @@ let
       cassandraYaml = builtins.toJSON cassandraConfigWithAddresses;
       cassandraEnvPkg = "${cfg.package}/conf/cassandra-env.sh";
       cassandraLogbackConfig = pkgs.writeText "logback.xml" cfg.logbackConfig;
+      passAsFile = [ "extraEnvSh" ];
+      inherit (cfg) extraEnvSh;
```

Comment:
This is necessary for `passAsFile`? Or could you do `passAsFile = [ "cfg.extraEnvSh" ];`?

Reply:
Yes, those two lines pretty much sum up how it works.

It's [documented](https://nixos.org/manual/nix/stable/#sec-advanced-attributes) in the Nix manual if you'd like to know more.

Reply:
Ahh, right! The second line is required to make it an environment variable in the first place.